api/python/quilt3/admin/__init__.py

```diff
@@ -0,0 +1,175 @@
+"""
+APIs for Quilt administrators. 'Registry' refers to Quilt stack backend services, including identity management.
+"""
+
+from datetime import datetime
+from typing import Any, Annotated, Literal, Union, Optional, List
+
+from pydantic import Field, TypeAdapter
+
+from ._graphql_client import *
+from ._graphql_client.base_model import UNSET, UnsetType
+
+
+class ManagedRole(BaseModel):
```

Comment:
I added these types to wrap what generated code returns, because otherwise we end up with 2 types for user, 3 types for managed roles and 3 types for unmanaged roles in signatures (and docs).

@nl0 
any better idea?

Reply:
> if nothing else, i guess you could just import the fragment types and use them directly -- they should be equivalent to the types you construct here (tho maybe names would be more noisy, but i guess you can just subclass them to fix this)

well, if I do something like
```python
class User(UserSelection):
    pass
```

1. it would still refer to `UserSelectionRoleUnmanagedRole`, `UserSelectionRoleManagedRole`, `UserSelectionExtraRolesUnmanagedRole` and `UserSelectionExtraRolesManagedRole`
2. field names end up spelled in camel case instead of snake case in type constructors and I think we should provide API with snake case

see https://github.com/quiltdata/quilt/blob/134c93747ed96098bd471262c4c550fc50bff9fd/api/python/quilt3/admin/_graphql_client/fragments.py#L41-L64

Reply:
another idea: since you're using custom classes anyways, just use plain dataclasses -- they will show up better (simpler?) in the docs. since the data is expected to be valid at that point, it's ok to trust it and not validate it again

Reply:
> since the data is expected to be valid at that point, it's ok to trust it and not validate it again

but I guess I would have to wrap nested non-scalars (e.g. `role` and `extra_roles`) manually?

Reply:
> but I guess I would have to wrap nested non-scalars (e.g. role and extra_roles) manually?

well, I think I don't

Reply:
ok, I switched to dataclasses, now doc looks a bit better